zip.go

```diff
@@ -187,8 +187,15 @@ func unzipAll(r *zip.Reader, destination string) error {
 }
 
 func unzipFile(zf *zip.File, destination string) error {
+	// to avoid zip slip (writing outside of the destination), we resolve
+	// the target path, and make sure it's nested in the intended
+	// destination, or bail otherwise.
+	destpath := filepath.Join(destination, zf.Name)
+	if !strings.HasPrefix(destpath, destination) {
+		return fmt.Errorf("%s: illegal file path", zf.Name)
+	}
```

Comment:
Do you think this logic could be pulled into its own function so that we don't have to repeat it everywhere? It's simple, I know, but it would be nice to standardize it.

Reply:
Took a stab at it. Let me know what you think.
I'm not sure about the name `sanitizeExtractPath`, so if you have any idea - that could be great.